chain/consensus/mir/mempool/handlers/batches.go

```diff
@@ -0,0 +1,84 @@
+package handlers
+
+import (
+	"github.com/filecoin-project/lotus/chain/consensus/mir/mempool/types"
+	"github.com/filecoin-project/mir/pkg/dsl"
+	mpdsl "github.com/filecoin-project/mir/pkg/mempool/dsl"
+	"github.com/filecoin-project/mir/pkg/pb/mempoolpb"
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+type State struct {
+	*types.State
+	NewTxIDs []t.TxID
+}
+
+// IncludeBatchCreation registers event handlers for processing NewRequests and RequestBatch events.
+func IncludeBatchCreation(
+	m dsl.Module,
+	mc *types.ModuleConfig,
+	params *types.ModuleParams,
+	commonState *types.State,
+) {
+	state := &State{
+		State:    commonState,
+		NewTxIDs: nil,
+	}
+
+	dsl.UponNewRequests(m, func(txs []*requestpb.Request) error {
+		mpdsl.RequestTransactionIDs(m, mc.Self, txs, &requestTxIDsContext{txs})
+		return nil
+	})
+
+	mpdsl.UponTransactionIDsResponse(m, func(txIDs []t.TxID, context *requestTxIDsContext) error {
+		for i := range txIDs {
+			state.TxByID[txIDs[i]] = context.txs[i]
+		}
+		state.NewTxIDs = append(state.NewTxIDs, txIDs...)
+		return nil
+	})
+
+	mpdsl.UponRequestBatch(m, func(origin *mempoolpb.RequestBatchOrigin) error {
```

Comment:
Was anything else except for this handler modified in this version of the mempool compared to Mir's default `/pkg/mempool/simplemempool`?

Reply:
Removed